### ISSUE TITLE
Does SMIL player only accept static SMIL XML file?

### ISSUE THREAD

**Issue by dynasigncorp** [signageos/smil-player#17]
I have created a dynamic link to generate a SMIL file from our content.   Here is the link:  https://pub.dynasign.net/ds5dev/get_smil.php?site_group=bpd&site=root_r1    
But somehow SMIL player does not like this link.    So I write the output from this link to a static file,  and put it here:   https://pub.dynasign.net/ds5dev/dynasign_smil_sample.xml
Now the SMIL player can accept this static link. 

Does that mean SMIL player can only play from the static link (filename ext. ends with .xml/.smil) ?

Below is content of this dynasign_smil_sample.xml
Thanks,

Eric Lu
--------------------------------------------------------
<?xml version="1.0" encoding="UTF-8"?>
<smil>
  <head>
    <layout>
      <root-layout width="1360" height="768"/>
      <region regionName="show1" left="0" top="0" width="875" height="658" z-index="1"/>
      <region regionName="show4" left="0" top="658" width="280" height="110" z-index="1"/>
      <region regionName="show6" left="875" top="0" width="485" height="658" z-index="1"/>
      <region regionName="marquee_message" left="280" top="658" width="1080" height="110" z-index="1"/>
    </layout>
  </head>
  <body>
    <par>
      <seq end="__prefetchEnd.endEvent">
        <seq repeatCount="indefinite">
          <video src="https://signageos-demo.s3.eu-central-1.amazonaws.com/smil/zones/files/loader.mp4"/>
        </seq>
      </seq>
      <par end="__prefetchEnd.endEvent" repeatCount="indefinite">
        <par>
          <excl>
            <priorityClass higher="stop" lower="defer" peer="stop">
              <par begin="wallclock(2008-07-21T11:36:00)" end="wallclock(2028-07-21T11:36:00)">
                <seq repeatCount="indefinite">
                  <img src="https://player.dynasign.net/dynasign/websites/bpd/base/dss_media/image/category/city/Austria/ViennaAustria1_1024x768.jpg" dur="10s" region="show1"/>
                </seq>
              </par>
            </priorityClass>
            <priorityClass higher="stop" lower="defer" peer="stop">
              <par begin="wallclock(2021-06-23T23:00:00)" end="wallclock(2099-12-31T23:59:00)">
                <seq repeatCount="indefinite">
                  <video src="https://player.dynasign.net/dynasign/websites/bpd/base/dss_media/video/hd_samples/VerizonRedHotDealDays_30s.mp4" dur="30s" region="show1"/>
                </seq>
              </par>
            </priorityClass>
          </excl>
        </par>
        <par>
          <excl>
            <priorityClass higher="stop" lower="defer" peer="stop">
              <par begin="wallclock(2021-06-24T00:00:00)" end="wallclock(2099-12-31T23:59:00)">
                <seq repeatCount="indefinite">
                  <img src="https://player.dynasign.net/dynasign/websites/bpd/base/dss_upload/root_r1/logoSmall180x76.gif" dur="10s" region="show4"/>
                </seq>
              </par>
            </priorityClass>
          </excl>
        </par>
        <par>
          <excl>
            <priorityClass higher="stop" lower="defer" peer="stop">
              <par begin="wallclock(2007-02-27T01:05:00)" end="wallclock(2051-01-01T00:59:00)">
                <seq repeatCount="indefinite">
                  <img src="https://player.dynasign.net/dynasign/websites/bpd/base/dss_upload/root/22bathaciendasm.jpg" dur="10s" region="show6"/>
                </seq>
              </par>
            </priorityClass>
          </excl>
        </par>
        <par>
          <excl/>
        </par>
      </par>
    </par>
  </body>
</smil>

**Comment by PetrDusek89** [comment]
Hello, 

on which device are you experiencing this issue? I tested link you provided on several devices and in emulator and its working fine.

I also noticed one more thing in your smil file.
This attribute is used only for intro definition, do not use it anywhere else.
`end="__prefetchEnd.endEvent"`

Thanks Petr

**Comment by dynasigncorp** [comment]
Did you mean you can play both  https://pub.dynasign.net/ds5dev/get_smil.php?site_group=bpd&site=root_r1  and
https://pub.dynasign.net/ds5dev/dynasign_smil_sample.xml  links?
I built and installed SMIL player on my Windows 10 PC.  The SMIL player has no response when I use this link, https://pub.dynasign.net/ds5dev/get_smil.php?site_group=bpd&site=root_r1,  on "SMIL file URL" field.   
It looks like SMIL player does not response the ENTER key?   

Then I installed  https://0099.io/8Zz4075/0.apk  on my Acer Android 6 tablet.   It also has the same behavior.
It does play https://pub.dynasign.net/ds5dev/dynasign_smil_sample.xml, but it has no response when I enter https://pub.dynasign.net/ds5dev/get_smil.php?site_group=bpd&site=root_r1.
Another issue I noticed is  it plays the whole video (on region regionName show1) at the beginning, but after that it plays one second of the video, the one second of the image, then loops video/image every second forever.  
This show1 region has two content, image duration 10 seconds, and video duration 30 seconds.  
Somehow SMIL player doesn't play the whole duration of each content?    Not sure whether you see this issue when you tried it?

Thanks,

Eric Lu

**Comment by PetrDusek89** [comment]
I tried both and both worked. This issue might be caused by incorrect build process.
Please upgrade to latest master and for build command use `npm run build-prod` instead of `npm run build --production`

Thanks Petr

**Comment by dynasigncorp** [comment]
You might be confused by the problems I have in different devices.

The issue I mentioned in this ticket is on my Android 6.0 tablet after installing  https://0099.io/8Zz4075/0.apk..

And the issues I have on my test locally in Windows 10 are very strange.
After running:   npm install && npm run prepare,  and  sos applet start ...   
Then try this http://192.168.1.252:8090  on the browser of this Windows 10,
I have different problems:   
The first one is the same one I had on Android 6.0.  It does not response to Enter key after I put in
https://pub.dynasign.net/ds5dev/get_smil.php?site_group=bpd&site=root_r1
 After few more tries by doing remove  .sosrc file and re-install, re-prepare,  I don't see that "SMIL file URL" field on the browser any more.  Instead it just redirects to different Applet from our account in signageos and shows the content that that Applet points to. 

Basically, I have created SMIL playlist, but I don't have reliable SMIL player to verify the playlist I created.

Thanks,

Eric Lu
